src/tools/pkcs11-tool.c

```diff
@@ -5048,6 +5103,155 @@ derive_ec_key(CK_SESSION_HANDLE session, CK_OBJECT_HANDLE key, CK_MECHANISM_TYPE
 #endif /* ENABLE_OPENSSL  && !OPENSSL_NO_EC && !OPENSSL_NO_ECDSA */
 }
 
+static CK_BBOOL s_true = TRUE;
+static CK_BBOOL s_false = FALSE;
+
+static void fill_attributes_seckey(CK_ATTRIBUTE *template, int *n_attrs, int max_attrs)
+{
+	assert(*n_attrs < max_attrs);
+	FILL_ATTR(template[*n_attrs], CKA_PRIVATE, opt_is_private?&s_true:&s_false, sizeof(CK_BBOOL));
+	(*n_attrs)++;
```

Comment:
Ideally, I would create a new macro which is checking that the template is large enough when a new attribute is added.

Reply:
Could the `FILL_ATTR` be extended to do this checking?